### PR TITLE
feat: add max row limit

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #3498 

### Description:

New env `LIGHTDASH_QUERY_MAX_LIMIT` that defaults to 5000

max row limit

Error for saved queries that exceed the limit.

Screenshot 2022-10-24 at 10 00 40

### CODE REVIEW

**Review by TuringLovesDeathMetal — APPROVED** [review]
nice one! looks good 😚 👌

File: packages/frontend/src/components/LimitButton/LimitForm.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import Input from '../ReactHookForm/Input';
+import { Props } from './index';
+import { ApplyButton, StyledForm } from './LimitButton.styles';
+
+const LimitForm: FC<Pick<Props, 'limit' | 'onLimitChange'>> = ({
+    limit,
+    onLimitChange,
+}) => {
+    const { health } = useApp();
+    const methods = useForm<{ limit: string }>({
```
Comment (rephus) [inline packages/frontend/src/components/LimitButton/LimitForm.tsx:16]: why is the limit a string and not a number ?

File: packages/frontend/src/components/LimitButton/LimitForm.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    if (!health.data) {
+        return null;
+    }
+    return (
+        <StyledForm
+            name="query_limit"
+            methods={methods}
+            onSubmit={handleSubmit}
+        >
+            <Input
```
Comment (rephus) [inline packages/frontend/src/components/LimitButton/LimitForm.tsx:45]: why not using a numeric input ?

File: packages/frontend/src/utils/fieldValidators.ts [hunk 3]
```diff
@@ -45,3 +45,8 @@ export const isValidEmailDomain: FieldValidator<string[]> =
                 : undefined;
         }
     };
+
+export const isOnlyNumbers: FieldValidator<string> = (fieldName) => (value) =>
+    !value || value.match(/\D/)
```
Comment (rephus) [inline packages/frontend/src/utils/fieldValidators.ts:50]: :+1:

**Review by rephus — APPROVED** [review]
Code looks good. Also great job with everything that is not code related:  adding env variable, asking the community..

File: packages/frontend/src/components/LimitButton/LimitForm.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        mode: 'onChange',
+    });
+    const {
+        setValue,
+        formState: { isValid },
+    } = methods;
+
+    useEffect(() => {
+        setValue('limit', limit?.toString());
+    }, [limit, setValue]);
```
Comment (IrakliJani) [inline packages/frontend/src/components/LimitButton/LimitForm.tsx:26]: form has `defaultValues` option that you can use instead

File: packages/frontend/src/components/LimitButton/LimitForm.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+    const handleSubmit = useCallback(
+        (data: { limit: string }) => {
+            console.log('handleSubmit');
```
Comment (IrakliJani) [inline packages/frontend/src/components/LimitButton/LimitForm.tsx:30]: log

**Review by IrakliJani — APPROVED** [review]
good stuff

File: packages/frontend/src/components/LimitButton/LimitForm.tsx (hunk 2 above)
Comment (ZeRego) [reply packages/frontend/src/components/LimitButton/LimitForm.tsx:45]: There is a really annoying behaviour when editing a number in the numeric input.
Eg: limit is 500 but you want 600. So the user deletes the 5 to then type 6. But the input instantly converts the `00` to 0 to be a valid number.

**Pull request merged**